controllers/spacebindingrequestmigration/spacebindingrequest_migration_controller.go

```diff
@@ -0,0 +1,183 @@
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, errs.Wrapf(err, "unable to get spacebinding")
+	}
+	if util.IsBeingDeleted(spaceBinding) {
+		logger.Info("the SpaceBinding is already being deleted")
+		return reconcile.Result{}, nil
+	}
+	// check if spaceBinding was created from SpaceBindingRequest,
+	// in that case we can skip it
+	if hasSpaceBindingRequest(spaceBinding) {
+		return reconcile.Result{}, nil
+	}
+
+	spaceName := types.NamespacedName{Namespace: spaceBinding.Namespace, Name: spaceBinding.Spec.Space}
+	space := &toolchainv1alpha1.Space{}
+	if err := r.Client.Get(ctx, spaceName, space); err != nil {
+		if errors.IsNotFound(err) {
+			// space was deleted
+			return reconcile.Result{}, nil
+		}
+		// error while reading space
+		return ctrl.Result{}, errs.Wrapf(err, "unable to get the bound Space")
+	}
+
+	murName := types.NamespacedName{Namespace: spaceBinding.Namespace, Name: spaceBinding.Spec.MasterUserRecord}
+	mur := &toolchainv1alpha1.MasterUserRecord{}
+	if err := r.Client.Get(ctx, murName, mur); err != nil {
+		if errors.IsNotFound(err) {
+			// mur was deleted
+			return reconcile.Result{}, nil
+		}
+		// error while reading MUR
+		return ctrl.Result{}, errs.Wrapf(err, "unable to get the bound MUR")
+	}
+
+	// skip workspace creator spacebinding
+	// the controller will convert only spacebindings created by system admins using the sandbox-cli.
+	// If the creator label on the space matches the usersignup reference on the MUR then this is the owner of the space
+	// and the spacebinding should not be migrated.
+	if mur.ObjectMeta.OwnerReferences == nil || len(mur.ObjectMeta.OwnerReferences) == 0 {
+		return ctrl.Result{}, errs.New("MasterUserRecord has no UserSignup owner reference")
+	}
+	usersignup := mur.ObjectMeta.OwnerReferences[0]
+	if space.Labels[toolchainv1alpha1.SpaceCreatorLabelKey] == usersignup.Name {
+		return reconcile.Result{}, nil
+	}
+
+	// get the spaceRole
+	spaceRole := spaceBinding.Spec.SpaceRole
+
+	// get member cluster name where the space was provisioned
+	targetCluster := space.Status.TargetCluster
+	memberCluster, memberClusterFound := r.MemberClusters[targetCluster]
+	if !memberClusterFound {
+		return ctrl.Result{}, errs.New(fmt.Sprintf("unable to find member cluster: %s", targetCluster))
+	}
+
+	// get the home namespace from space
+	defaultNamespace := getDefaultNamespace(space.Status.ProvisionedNamespaces)
+
+	// create the SpaceBindingRequest resource
+	sbrName := mur.GetName() + "-" + spaceRole
+	sbr := &toolchainv1alpha1.SpaceBindingRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      sbrName,
+			Namespace: defaultNamespace,
+			Labels: map[string]string{
+				toolchainv1alpha1.SpaceCreatorLabelKey:                 "spacebindingrequestmigration",
```

Comment:
I wonder if it can cause issues in the future if we use a non-existing username as the create label... Maybe it will be safer to use the space owner as the creator?

WDYT?

Reply:
Yeah I was unsure about this too. I added something custom just to be able to distinguish the SpaceBindingRequests we created via this controller from the "regular" ones. Maybe I can use another label , and not   `SpaceCreatorLabelKey` ?

Reply:
We could use another label for that but do we really need to make it even more complicated? IMO it should be fine if we create these SBRs "on the behalf of" the owner.
@MatousJobanek WDYT?

Reply:
let's just drop the label - it's not used in SBRs anywhere. 
```suggestion
```

Reply:
no , it is not used. The main usage was potentially for manual investigation. Dropped. Thanks!